idr/marshal1.go

```diff
@@ -0,0 +1,54 @@
+package idr
+
+import (
+	"fmt"
+
+	"github.com/jf-tech/go-corelib/jsons"
+)
+
+// j1NodePtrName returns a categorized name for a *Node pointer used in JSONify1
+func j1NodePtrName(n *Node) string {
+	if n == nil {
+		return "(nil)"
+	}
+	name := func(n *Node) string {
+		if IsXML(n) && XMLSpecificOf(n).NamespacePrefix != "" {
+			return XMLSpecificOf(n).NamespacePrefix + ":" + n.Data
+		}
+		return n.Data
+	}
+	switch n.Type {
+	case DocumentNode:
+		return fmt.Sprintf("(%s)", n.Type)
+	case ElementNode, AttributeNode:
+		return fmt.Sprintf("(%s %s)", n.Type, name(n))
+	case TextNode:
+		return fmt.Sprintf("(%s '%s')", n.Type, n.Data)
+	default:
+		return fmt.Sprintf("(unknown '%s')", n.Data)
+	}
+}
+
+// j1NodeToInterface converts *Node into an interface{} suitable for json marshaling used in JSONify1.
+func j1NodeToInterface(n *Node) interface{} {
```

Comment:
Wondering whether we need to consider n is `nil` for this function. Do we have UT to cover `nil` situation for `JSONify1` or `j1NodeToInterface`?

Reply:
no. these are not in a sense public API, nil is not guarded.